src/main/java/it/gov/pagopa/payhub/activities/service/CsvService.java

```diff
@@ -70,6 +70,62 @@ public void createCsv(Path csvFilePath, List<String[]> header, List<String[]> da
         log.info("CSV file created successfully: {}", csvFilePath);
     }
 
+    /**
+     * Creates a CSV file from the provided supplier of beans.
+     *
+     * <p>This method ensures that the file is properly closed after processing
+     * by using a try-with-resources statement.</p>
+     *
+     * <p>This method uses {@code StatefulBeanToCsv.write()}
+     * to write each bean individually, reducing memory consumption at the cost of lower performance.</p>
+     *
+     * <p>The supplier is called repeatedly until it returns an empty list or null,
+     * indicating that there are no more beans to write.</p>
+     *
+     * @param <C> the generic type of the beans to be written to the CSV
+     * @param csvFilePath the path to the CSV file to write
+     * @param typeClass the class type of the beans to be written to the CSV
+     * @param csvRowsSupplier a supplier of beans to be written to the CSV (called multiple times)
+     * @param csvProfile the profile to be used for writing the CSV
+     * @throws IOException if an error occurs while writing the file
+     */
+    public <C> void createCsv(Path csvFilePath, Class<C> typeClass, Supplier<List<C>> csvRowsSupplier, String csvProfile) throws IOException {
+        if (csvFilePath == null || csvRowsSupplier == null) {
+            throw new IllegalArgumentException("Arguments cannot be null");
+        }
```

Comment:
i would neither check this... if this happen, even without these lines, you will implicilty throw NullPointerException...
You are not catching the IllegalArgument elsewhere, thus there is not reason to handle explicitly)